packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,156 @@
+      threeParticipantJoinedNoticeString: locale.threeParticipantJoinedNoticeString,
+      participantLeftNoticeString: locale.participantLeftNoticeString,
+      twoParticipantLeftNoticeString: locale.twoParticipantLeftNoticeString,
+      threeParticipantLeftNoticeString: locale.threeParticipantLeftNoticeString,
+      unnamedParticipantChangedString: locale.unnamedParticipantChangedString,
+      manyParticipantsChanged: locale.manyParticipantsChanged,
+      manyUnnamedParticipantsChanged: locale.manyUnnamedParticipantsChanged
+    };
+  }, [locale]);
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  const remoteParticipants = useSelector(getRemoteParticipantsConnectedSelector);
+  const [previousParticipants, setPreviousParticipants] = useState<RemoteParticipantState[]>(remoteParticipants);
+
+  useEffect(() => {
+    const setParticipantEventString = (string: string): void => {
+      setAnnouncerString('');
+      setAnnouncerString(string);
+    };
+
+    if (previousParticipants.length > remoteParticipants.length) {
+      const whoLeft = previousParticipants.filter((p) => !remoteParticipants.includes(p));
+      //someone left
+      setParticipantEventString(createAnnouncmentString('left', whoLeft, strings));
+      setPreviousParticipants(remoteParticipants);
+    } else if (remoteParticipants.length > previousParticipants.length) {
+      const whoJoined = remoteParticipants.filter((p) => !previousParticipants.includes(p));
+      // someone joined
+      setParticipantEventString(createAnnouncmentString('joined', whoJoined, strings));
+      setPreviousParticipants(remoteParticipants);
+    }
+  }, [remoteParticipants, previousParticipants, strings]);
+
+  return announcerString;
+};
+
+/**
+ * Generates the announcement string for when a participant joins or leaves a call.
+ */
+export const createAnnouncmentString = (
+  direction: 'joined' | 'left',
+  participants: RemoteParticipantState[],
+  strings: ParticipantChangedAnnouncmentStrings
+): string => {
+  /**
+   * Check that we have more than 1 participant, if they all have no displayName return unnamed participants
+   * overflow string
+   */
+  if (participants.filter((p) => p.displayName).length === 0 && participants.length > 0) {
+    const participantsString = _formatString(strings.manyUnnamedParticipantsChanged, {
+      numOfParticipants: (participants.length - 1).toString()
+    });
+    return _formatString(
+      direction === 'joined' ? strings.participantJoinedNoticeString : strings.participantLeftNoticeString,
+      { displayNames: participantsString }
+    );
+  }
+  const oneName = participants[0].displayName ?? strings.unnamedParticipantChangedString;
+  const twoNames = participants.map((p) => p.displayName ?? strings.unnamedParticipantChangedString);
+  const threeNames = participants.map((p) => p.displayName ?? strings.unnamedParticipantChangedString);
```

Comment:
nit: twoNames and threeNames do exactly the same thing, and one name functionally does the same thing.
Consider 
```
const participantNames = participants.map((p) => p.displayName ?? strings.unnamedParticipantChangedString);

switch (participantNames.length) {
  case 0: return;
  case 1: ...  { displayName: participantNames[0] }
  case 2: ...  { displayName1: participantNames[0], participantNames[1] }
  case 3: ... { displayName1: participantNames[0], participantNames[1], participantNames[2] }
  default: // many participants
    ... { displayName1: participantNames[0], participantNames[1], participantNames[2], numberOfExtraParticipants }
```